src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1422,6 +1421,12 @@ public static bool IsInstanceContext(this SyntaxTree syntaxTree, int position, S
                 return false;
             }
 
+            var expressionBody = GetExpressionBodyFromMember(containingMember);
+            if (expressionBody != null)
+            {
+                return ArrowExpressionClauseExpressionAdjustedSpan(expressionBody).IntersectsWith(position);
```

Comment:
I believe he's using the span to avoid having to walk up the tree.

Consider `int M => ((this.x + this.y) * $$)` -- you'd have to walk up the tree until you found an ExpressionSyntax that was the Expression child of the ArrowExpresionClause, no?